tensorflow/contrib/distribute/python/mirrored_strategy.py

```diff
@@ -615,7 +615,8 @@ def _get_cross_tower_ops(self):
     return self._cross_tower_ops
 
   def _reduce(self, aggregation, value, destinations):
-    assert not isinstance(value, values.Mirrored)
+    if isinstance(value, values.Mirrored):
+      return value
```

Comment:
You need to check whether the aggregation is mean and `isinstance(value, values.Mirrored)` is True. If yes, return `value`, otherwise run the old code path.

Reply:
Thanks! Can we still keep `assert not isinstance(value, values.Mirrored)` after this logic? We prefer to format the code like:
```
if (isinstance(value, values.Mirrored) and
    aggregation == variable_scope.VariableAggregation.MEAN)):
    return value
```